lib/utilities/package-cache.js

```diff
@@ -0,0 +1,524 @@
+ * directory with no change, making the following invocation simply a
+ * cache validation check:
+ *
+ * ```
+ * var dir2 = cache.create('your-cache', 'yarn', '{
+ *   "dependencies": {
+ *     "lodash": "*",
+ *     "ember-cli": "*"
+ *   }
+ * }');
+ * // => /somewhere/tmp/your-cache-A3B4C6D
+ * ```
+ *
+ * If you wish to modify a cache you can do so using the `update` API:
+ *
+ * ```
+ * var dir3 = cache.update('your-cache', 'yarn', '{
+ *   "dependencies": {
+ *     "": "*",
+ *     "lodash": "*",
+ *     "ember-cli": "*"
+ *   }
+ * }');
+ * // => /somewhere/tmp/your-cache-A3B4C6D
+ * ```
+ *
+ * Underneath the hood `create` and `update` are identical–which
+ * makes clear the simplicity of this tool. It will always do the
+ * right thing. You can think of the outcome of any `create` or
+ * `update` call as identical to `rm -rf node_modules && npm install`
+ * except as performant as possible.
+ *
+ * If you need to make modifications to a cache but wish to retain
+ * the original you can invoke the `clone` command:
+ *
+ * ```
+ * var newDir = cache.clone('your-cache', 'modified-cache');
+ * var manifest = fs.readJsonSync(path.join(newDir, 'package.json'));
+ * manifest.dependencies['express'] = '*';
+ * cache.update('modified-cache', 'yarn', JSON.stringify(manifest));
+ * // => /somewhere/tmp/modified-cache-F8D5C8B
+ * ```
+ *
+ * This mental model makes it easy to prevent coding mistakes, state
+ * leakage across multiple test runs by making multiple caches cheap,
+ * and has tremendous performance benefits.
+ *
+ * You can even programatically update a cache:
+ *
+ * ```
+ * var CommandGenerator = require('./command-generator');
+ * var yarn = new CommandGenerator(require.resolve('yarn/bin/yarn.js'), {
+ *   retryCommands: ['install', 'upgrade']
+ * });
+ *
+ * var dir = cache.create('your-cache', 'yarn', "{ ... }");
+ *
+ * yarn.invoke('add', 'some-addon', { cwd: dir });
+ * ```
+ *
+ * The programmatic approach enables the entire set of usecases that
+ * the underlying package manager supports while continuing to wrap it
+ * in a persistent cache. You should not directly modify any files in the
+ * cache other than the manifest as that puts the cache unless you really
+ * know what you're doing as that can put the cache into a possibly
+ * invalid state.
+ *
+ * As the only caveat, PackageCache _is_ persistent. The consuming
+ * code is responsible for ensuring that the cache size does not
+ * grow unbounded.
+ *
+ * @class PackageCache
+ * @constructor
+ * @param {Object} options Options hash. Supports:
+ * @param {Boolean} [options.linkEmberCLI=false] If after installation should link Ember CLI.
+ */
+function PackageCache(options) {
+  this._conf = new Configstore('package-cache');
+  this.options = merge({
+    linkEmberCLI: false
+  }, options);
+
+  // Wrap the Configstore in a pretty interface.
+  Object.defineProperty(this, 'dirs', {
+    get: function() {
+      return this._conf.all;
+    },
+    set: function() {
```

Comment:
I went with this to make the error a bit more useful, don't feel too strongly though. Removed.